packages/react-instantsearch/src/core/indexUtils.js

```diff
@@ -125,13 +133,25 @@ export function getCurrentRefinementValue(
   refinementsCallback
 ) {
   const index = getIndex(context);
+  const namespace = getNamespaceAndId(id);
   const refinements =
     (hasMultipleIndex(context) && has(searchState, `indices.${index}.${id}`)) ||
+    (!hasMultipleIndex(context) &&
+      namespace.namespace &&
+      namespace.id &&
+      has(searchState[namespace.namespace], namespace.id)) ||
     (!hasMultipleIndex(context) && has(searchState, id));
```

Comment:
This is not needed anymore right? We can handle only two case (one for multi indices, one regular) no?